### PR TITLE
add document property

### PR DESCRIPTION
**Pull request by Simon-Laux**
counterpart of https://github.com/webxdc/webxdc-dev/pull/28

### CODE REVIEW

File: webxdc.d.ts [hunk 1]
```diff
@@ -24,7 +27,10 @@ type ReceivedStatusUpdate<T> = {
   max_serial: number;
   /** optional, short, informational message. */
   info?: string;
-  /** optional, short text, shown beside the webxdc's icon. */
+  /** optional, if the Webxdc creates a document, this is the name of the document;
+   * not set if the Webxdc does not create a document */
+  document?: string;
+  /** optional, short text, shown beside app icon. */
```
Comment (Simon-Laux) [inline webxdc.d.ts:33]:
```suggestion
  /** optional, short text, shown beside the webxdc's icon. */
```

File: webxdc.d.ts [hunk 2]
```diff
@@ -9,7 +9,10 @@ type SendingStatusUpdate<T> = {
    * usually only one line of text is shown,
    * use this option sparingly to not spam the chat. */
   info?: string;
-  /** optional, short text, shown beside the icon;
+  /** optional, if the Webxdc creates a document, you can set this to the name of the document;
+   * do not set if the Webxdc does not create a document */
+  document?: string;
+  /** optional, short text, shown beside app icon;
```
Comment (Simon-Laux) [inline webxdc.d.ts:15]:
```suggestion
  /** optional, short text, shown beside icon;
```

**Review by r10s — APPROVED** [review]

File: webxdc.d.ts [hunk 3]
```diff
@@ -9,7 +9,10 @@ type SendingStatusUpdate<T> = {
    * usually only one line of text is shown,
    * use this option sparingly to not spam the chat. */
   info?: string;
-  /** optional, short text, shown beside the icon;
+  /** optional, if the Webxdc creates a document, you can set this to the name of the document;
+   * do not set if the Webxdc does not create a document */
+  document?: string;
+  /** optional, short text, shown beside icon;
```
Comment (Simon-Laux) [inline webxdc.d.ts:15]:
```suggestion
  /** optional, short text, shown beside the icon;
```

File: webxdc.d.ts (hunk 1 above)
Comment (r10s) [reply webxdc.d.ts]: i'd prefer this over the other variant.

Comment (Simon-Laux) [reply webxdc.d.ts]: back to app?

Comment (r10s) [reply webxdc.d.ts]: no :)

i meant that the wording "beside the icon" from the other change is a bit less clear than this one. but both are fine, and repetition is also an issue :) so, everyrhing's fine. sorry for the noise.

**Review by adbenitez — APPROVED** [review]

**Pull request merged**